clientconn_state_transition_test.go

```diff
@@ -388,6 +397,19 @@ func TestStateTransitions_MultipleAddrsEntersReady(t *testing.T) {
 	}
 }
 
+// Keep reading until something causes the connection to die (EOF, server closed, etc). Useful
+// as a tool for mindlessly keeping the connection healthy, since the client will error if
+// things like client prefaces are not accepted in a timely fashion.
+func keepReading(conn net.Conn) {
+	buf := make([]byte, 1024)
+	for {
```

Comment:
Code golf:

```go
for err := nil; err == nil; _, err = conn.Read(buf) {}
```

Reply:
Nice haha. Done, kind of: err needs a type, so `for _, err := conn.Read`.

Reply:
That works.  Or `err := error(nil)` would work to avoid the repeated `Read` invocation.